src/BuildingRegistry/Building/Commands/ReaddressAddresses.cs

```diff
@@ -0,0 +1,66 @@
+namespace BuildingRegistry.Building.Commands
+{
+    using System;
+    using System.Collections.Generic;
+    using Be.Vlaanderen.Basisregisters.Generators.Guid;
+    using Be.Vlaanderen.Basisregisters.GrAr.Provenance;
+    using Be.Vlaanderen.Basisregisters.Utilities;
+
+    public class ReaddressAddresses : IHasCommandProvenance
+    {
+        private static readonly Guid Namespace = new Guid("444c245f-b4ab-463d-bbe2-c88769953481");
+        public BuildingPersistentLocalId BuildingPersistentLocalId { get; }
+        public IReadOnlyDictionary<BuildingUnitPersistentLocalId, IReadOnlyList<ReaddressData>> Readdresses { get; }
+        public Provenance Provenance { get; }
+
+        public ReaddressAddresses(
+            BuildingPersistentLocalId buildingPersistentLocalId,
+            IReadOnlyDictionary<BuildingUnitPersistentLocalId, IReadOnlyList<ReaddressData>> readdresses,
+            Provenance provenance)
+        {
+            BuildingPersistentLocalId = buildingPersistentLocalId;
+            Readdresses = readdresses;
+            Provenance = provenance;
+        }
+
+        public Guid CreateCommandId()
+            => Deterministic.Create(Namespace, $"ReaddressAddresses-{ToString()}");
+
+        public override string? ToString()
+            => ToStringBuilder.ToString(IdentityFields());
+
+        private IEnumerable<object> IdentityFields()
+        {
+            yield return BuildingPersistentLocalId;
+
+            foreach (var address in Readdresses)
```

Comment:
Shouldn't variable name be buildingUnit instead of address?

Reply:
Done